tripy/tripy/frontend/trace/ops/binary_elementwise.py

```diff
@@ -167,11 +167,22 @@ def to_flat_ir(self, inputs, outputs):
 @TENSOR_METHOD_REGISTRY("__add__")
 @TENSOR_METHOD_REGISTRY("__radd__")
 @frontend_utils.convert_inputs_to_tensors(sync_arg_types=[("self", "other")])
-def __add__(self, other: Union["tripy.Tensor", Any]) -> "tripy.Tensor":
+@dtype_info.dtype_info(
+    dtype_variables={"T1": ["float32", "float16", "bfloat16", "int8", "int32", "int64", "bool"]},
+    dtype_constraints={"self": "T1", "other": "T1", dtype_info.RETURN_VALUE: "T1"},
+)
+@dtype_info.dtype_info(
+    dtype_variables={"T1": ["float32", "float16", "bfloat16", "int8", "int32", "int64", "bool"]},
+    dtype_constraints={"other": "T1", dtype_info.RETURN_VALUE: "T1"},
+    param_type_specification={"self": numbers.Number},
+    function_name="__radd__",
+)
```

Comment:
I'm not sure this is actually meaningful for any of the `__r<op>__` methods since the left operand is always a non-tripy tensor, so data type isn't really meaningful. Maybe we can drop the type constraints for those methods? That would also let us drop the extra logic for `function_name`

Reply:
The right hand side still takes dtype and the output has sometime of dtype, so I think it might be still helpful to have it. I will test it out and see the results of removing type constraints on the documentation and let you know when I have an update.

Reply:
That's true, I guess it's still useful for showing how the output type relates to the inputs.

Reply:
Ill revert the commit then.